release/src/router/rc/wanduck.c

```diff
@@ -2352,10 +2352,11 @@ void handle_dns_req(int sfd, unsigned char *request, int maxlen, struct sockaddr
 		return;
 	opcode = d_req->flag_set.flag_num & htons(0x7800);
 	ptr = request + sizeof(dns_header);
-	end = request + maxlen;
+	/* queries.name is PATHLEN long, one trailing byte for null */
+	end = request + (maxlen > PATHLEN-1 ? PATHLEN-1 : maxlen);
 
 	/* query, only first so far */
-	memset(&queries, 0, sizeof(queries));
+	memset(&queries, 0, sizeof(dns_queries));
```

Comment:
quite valid, but what if sizeof(valiable) here is safer than sizeof(typeof(variable))

Reply:
Doing so increase the possibility of zero'ing not enough bytes if the struct contains padding. That being said, sizeof(variable) is preferable when doing arithmetic.

This is typically more significant when trying to erase sensitive data, for this project it likely won't cause an issue.

Reply:
if sizeof of struct & variable doesn’t match there should be something very wrong with compiler. using one or another approach, I think, depends on exact case, i.e while ops on variable - preferable use sizeof(variable), while allocating struct - malloc(sizeof(struct)) although sizeof(*variable) is still valid.
as for cleaning struct members incl. possible padding , there’s no other way than to zero from this member start to next member start. not the case here, the whole struct is zeroed.
so.. matter of taste, in the end :)

Reply:
Hmm..I think you're right in that sizeof(variable) is safer. We might later change the type.
I'm probably thinking about sizeof(variable) not necessarily equals sizeof(variable.a) + sizeof(variable.b)...etc.